pulse2percept/models/beyeler2019.py

```diff
@@ -774,21 +781,26 @@ def _predict_spatial(self, earray, stim):
             warnings.warn(msg)
         # This does the expansion of a compact stimulus and a list of
         # electrodes to activation values at X,Y grid locations:
-        if self.engine != 'jax':
+        if self.engine in ['serial', 'cython']:
             return fast_axon_map(stim.data,
-                                 np.array([earray[e].x for e in stim.electrodes],
-                                          dtype=np.float32),
-                                 np.array([earray[e].y for e in stim.electrodes],
-                                          dtype=np.float32),
-                                 self.axon_contrib,
-                                 self.axon_idx_start.astype(np.uint32),
-                                 self.axon_idx_end.astype(np.uint32),
-                                 self.rho,
-                                 self.thresh_percept,
-                                 self.n_threads)
-        else:
+                                np.array([earray[e].x for e in stim.electrodes],
+                                        dtype=np.float32),
+                                np.array([earray[e].y for e in stim.electrodes],
+                                        dtype=np.float32),
+                                self.axon_contrib,
+                                self.axon_idx_start.astype(np.uint32),
+                                self.axon_idx_end.astype(np.uint32),
+                                self.rho,
+                                self.thresh_percept,
+                                self.n_threads)
+        elif self.engine == 'torch':
+            e_locs = torch.tensor([(x,y) for x,y in zip(np.array([earray[e].x for e in stim.electrodes], dtype=np.float32),
+                                                        np.array([earray[e].y for e in stim.electrodes], dtype=np.float32))]).to(self.device)
```

Comment:
Also minor, but prefer to use torch.tensor(..., device=device) instead of torch.tensor(...).to(device) to avoid unnecessary copy